Match.cpp

```diff
@@ -331,21 +331,115 @@ void Match::print( ostream& fout )
 		fout.write( (char*) &hom[1] , sizeof( bool ) );
 	} else
 	{
-		fout << node[0]->getID() << '\t';
-		fout << node[1]->getID() << '\t';
-		fout << ALL_SNPS.getSNP(snp_start).getChr() << '\t';
-		fout << ALL_SNPS.getSNP(snp_start).getPhysPos() << ' ';
-		fout << ALL_SNPS.getSNP(snp_end).getPhysPos() << '\t';
-		fout << ALL_SNPS.getSNP(snp_start).getSNPID() << ' ';
-		fout << ALL_SNPS.getSNP(snp_end).getSNPID() << '\t';
-		fout << ( snp_end - snp_start + 1) << '\t';
-		fout << setiosflags(ios::fixed) << setprecision(2) << distance << '\t';
-		if ( genetic ) fout << "cM" << '\t'; else fout << "MB" << '\t';
-		fout << dif;
-		for ( int n = 0 ; n < 2 ; n++ )
-			if ( hom[n] ) fout << '\t' << 1; else fout << '\t' << 0;
-		fout << endl;
+		if ( ALL_SAMPLES.useEmbarkRFGermlineOutput ) {
+			int key1 = stoi(node[0]->single_id);
+			int key2 = stoi(node[1]->single_id);
+			vector<string> oline;
+			string joined_oline;
+			ofstream* ofs;
+			string chromosome = ALL_SNPS.getSNP(snp_start).getChr();
+			string start_pos = to_string(ALL_SNPS.getSNP(snp_start).getPhysPos());
+			string end_pos = to_string(ALL_SNPS.getSNP(snp_end).getPhysPos());
+			
+			// homoz
+			if ( key1 == key2 && node[0]->is_new ) {
```

Comment:
You could DRY out these if statements significantly, which I think would improve readability. The only thing that changes between the branches is `ofs`, and whether we get the 0th or 1st element of node. You could just set these in an if statement, then all the oline pushes could be shared

Reply:
new dog new dog has some extra logic though which you'll need to keep in the if statements (maybe this is why you didn't DRY this out?)

Reply:
I didn't do this mostly because I wanted the logic to be very visible and easily comparable to the logic in the rust code